cigaradvisor/blocks/article-teaser/article-teaser.js

```diff
@@ -13,21 +13,28 @@ function formatDate(originalDateString) {
   return `${formattedDate}|${dateTimeAttribute}`;
 }
 
-export default async function decorate(block) {
-  const filterPath = block.querySelector('a').getAttribute('href');
-  block.classList.add('article-teaser');
-  const articleInfo = await fetchData(getRelativePath(filterPath), '/cigaradvisor/posts/query-index.json');
-  if (!articleInfo) {
-    return;
+// eslint-disable-next-line max-len
+export async function buildArticleTeaser(parentElement, articleInfo, articleCategoryInfo, articleAuthorInfo) {
```

Comment:
Rather than have this method check for all the values. Have the it assume that everything will be provided - you can define the structure as a JSDoc, it can even be the exact entry in the Query index(es)

Then the `decorate` function is responsible for fetching all the data, this function just renders it.

Reply:
when we are on category page, articleinfo and articleCategoryInfo will be provided to this function only authorinfo needs to be fetched.

when on an author page, articleinfo and authorinfor will be provided to this function only categoryinfo for each article needs to be fetched from the default query-index.

Reply:
I'm suggesting that this function should enforce that _all_ of this information should be passed to it, regardless of which block is calling it